File: sh2.java
import java.util.ArrayList;

public class sh2{
  public static void main(String[] args) {
    ArrayList<Double> theList = new ArrayList<Double>();
    theList.add(1.5);
    theList.add(2.0);
    theList.add(3.2);
    theList.add(4.8);
    System.out.println(sumList(theList));
  }
  static double sumList(ArrayList<Double> list){
    double sum = 0.00;
    for (int i = 0; i < list.size(); i++) {
          sum += list.get(i);
        }
    return sum;
  }
}

File: sh2.cpp
#include <iostream>
#include <vector>
double sumList(const std::vector<double>& list) {
    double sum = 0.00;
    for (int i = 0; i < list.size(); i++) {
        sum += list[i];
    }
    return sum;
}
int main() {
    std::vector<double> theList;
    theList.push_back(1.5);
    theList.push_back(2.0);
    theList.push_back(3.2);
    theList.push_back(4.8);

    std::cout << sumList(theList) << std::endl;

    return 0;
}